packages/protocol/engine/consensus/acceptance/gadget.go

```diff
@@ -120,15 +165,33 @@ func (a *Gadget) RefreshSequenceAcceptance(sequenceID markers.SequenceID, newMax
 		marker := markers.NewMarker(sequenceID, markerIndex)
 
 		markerVoters := a.tangle.VirtualVoting.MarkerVoters(marker)
-		if a.tangle.ValidatorSet.IsThresholdReached(markerVoters.TotalWeight(), a.optsMarkerAcceptanceThreshold) && a.setMarkerAccepted(marker) {
-			queuedBlocks = append(queuedBlocks, a.propagateAcceptance(marker)...)
+
+		if validator.IsThresholdReached(totalWeight, a.tangle.ValidatorSet.TotalWeight(), a.optsConfirmationThreshold) {
+			// have enough weight to confirm based on total weight
+			if validator.IsThresholdReached(totalWeight, markerVoters.TotalWeight(), a.optsConfirmationThreshold) {
```

Comment:
This could use a little bit more explanation or we could move it into a utility function `tryConfirm` or something

Reply:
Moved the whole if to a separate function with a comment and added more inline comments.